src/components/pages/DetailsPage/DetailsPage.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import style from './DetailsPage.module.scss';
+import { useNavigate, useParams } from 'react-router-dom';
+import { ROUTER_PATHS, URL } from '../../../models/enums';
+import { ShowData } from '../../../models/interfaces';
+import { AiFillStar } from 'react-icons/ai';
+import { GoLinkExternal } from 'react-icons/go';
+import Spinner from '../../Spinner/Spinner';
+
+interface DetailsPageProps {
+  setIsCardItemsDarked: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const DetailsPage: React.FC<DetailsPageProps> = ({ setIsCardItemsDarked }) => {
+  const { id } = useParams();
+
+  const navigate = useNavigate();
+
+  const goBack = () => {
+    navigate(`${ROUTER_PATHS.MAIN}`);
+    setIsCardItemsDarked(false);
+  };
+
+  const noDate = 'no data to show';
+
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [show, setShow] = React.useState({
+    id: 0,
+    image: '',
+    name: '',
+    rating: 0,
+    language: '',
+    summary: '',
+    genres: [noDate],
+    premiered: '',
+    ended: '',
+    officialSite: '',
+  });
+
+  const getShowById = async () => {
+    if (isLoading) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      const response = await fetch(`${URL.SHOWS}/${id}`);
+
+      if (!response.ok) {
+        throw new Error('Network response error');
+      }
+
+      const data: ShowData = await response.json();
+
+      show.genres = [];
+      const image = data.image?.original || '';
+
+      setShow({
+        id: data.id,
+        image,
+        name: data.name,
+        rating: data.rating.average,
+        language: data.language,
+        summary: data.summary,
+        genres: data.genres || [noDate],
+        premiered: data.premiered,
+        ended: data.ended,
+        officialSite: data.officialSite,
+      });
+
+      setIsLoading(false);
+    } catch (error) {
+      console.error(error);
+      setIsLoading(false);
+    }
+  };
+
+  React.useEffect(() => {
+    getShowById();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
+
+  const years = `${show.premiered ? show.premiered.slice(0, 4) : ''} ${
+    show.ended ? '- ' + show.ended.slice(0, 4) : ''
+  } `;
+
+  return (
```

Comment:
Consider to break down into more simple separate components

Reply:
done